@xen-orchestra/backups/_runners/_writers/_MixinXapiWriter.js

```diff
@@ -14,6 +14,19 @@ exports.MixinXapiWriter = (BaseClass = Object) =>
       this._sr = sr
     }
 
+    // check if the base Vm has all its disk on health check sr
+    async #isAlreadyOnHealthCheckSr(baseVm) {
+      const xapi = baseVm.$xapi
+      let onSameSr = true
+      const vdiRefs = await xapi.VM_getDisks(baseVm.$ref)
+
+      for (const vdiRef of vdiRefs) {
+        const vdi = xapi.getObject(vdiRef)
+        onSameSr = onSameSr && vdi.$SR.uuid === this._healthCheckSr.uuid
+      }
+      return onSameSr
```

Comment:
I find this approach simpler:

```suggestion
      const vdiRefs = await xapi.VM_getDisks(baseVm.$ref)
      for (const vdiRef of vdiRefs) {
        const vdi = xapi.getObject(vdiRef)
        if (vdi.$SR.uuid !== this._heathCheckSr.uuid) {
          return false
        }
      }
      return true
```